### DISCUSSION TITLE
How to load my own javascript

### DISCUSSION THREAD

**Discussion by rx65m** [dimdenGD/OldTwitter#686] [Q&A]
Hello dimden and everyone else.
On regular X-Twitter I inject a Tampermonkey script to filter certain tweets, like the ones I have already reposted or the ones with videos, among some other functions.
Although, I can see my Tampermonkey script buttons loaded when the X-Twitter page opens, unfortunately once this fantastic extension of yours gets loaded, my Tampermonkey script gets lost so I can't use it together with OldTwitter.
I wonder if I'm allowed to load my own javascript together with the OldTwitter ones, and if so, what are the steps I must follow to respect your amazing work, and to be sure my OldTwitter keeps running smoothly.
Thanks everyone in advance.
Rex.

**Comment by Nexxkinn** [comment]
Have the same problem with my own userscripts too.
My current mitigation so far is to keep reloading the page betting for the race condition where violentmonkey injects it fast enough before OT takes over the page.

I'd be very glad if there's a solution to load my userscript on the extension by e.g. giving an option on the OT settings to add your own userscript or something.

**Comment by dimdenGD** [comment]
I already tried to code plugins support, but unfortunately it's not really possible with way it's written because of Manifest V3 limitations. I'm sure it's possible to make userscript loads consistent though, instead of starting script as soon as possible you should detect whatever OldTwitter elements appearing, that would mean OldTwitter injected and it's a good time to start the script.

**Comment by rx65m** [comment]
Thank you very much for your kind reply.
I understand what you mean.
Please, allow me to ask you 2 different questions if you don't mind:
• Is it possible to save my javascript file in one of the folders of your amazing extension, to be loaded together with the rest of your scripts? If yes, what folder would you recommend, and what other modifications should I do to be sure that my javascript file gets loaded as well.
• What element would you recommend me to observe as a trigger to inject my javascript code using Tampermonkey or any other javascript inyector?

**Comment by dimdenGD** [comment]
scripts folder probably, add script to scripts/injection.js
`#center-cell` for example

**Comment by rx65m** [comment]
Thank you very much.
Very kind from you to take the time to attend my messages.

**Comment by Nexxkinn** [comment]
Thanks, now it works properly in my case by implementing these:
- `@run-at   document-start`
- `@inject-into  page`
- add initial step to wait until OT extension is loaded
- inject some css styling with `GM_addStyle()` when loaded
- run main userscript when loaded

**Comment by rx65m** [comment]
Adding different text to original OldTwitter files, or adding different files to any folder of it, makes the extension get uninstalled.
So that option is not working unless I learn how to force the extension to accept its edited files.

Now I'm using "body.body-old-ui.body-light.body-dark.body-pitch-black" as my trigger to load my Tampermonkey script, but I still need to reload each new page to see my buttons.
I just can't make my javascript to get loaded during the first loading of the page. Hence, X-Twitter is penalizing me for re-loading too many times, and after 6 or 7 pages I can't load any other page before 1 hour.
